### PR TITLE
Feature/refactor metadata

### PR DESCRIPTION
**Pull request by artem-v**

### CODE REVIEW

**Comment by ronenhamias** [comment]
this is a breaking change?

File: cluster/src/main/java/io/scalecube/cluster/metadata/MetadataStoreImpl.java [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+    // Stop accepting requests
+    actionsDisposables.dispose();
+    // Clear metadata hash map
+    membersMetadata.clear();
+  }
+
+  @Override
+  public Map<String, String> metadata() {
+    // Get local metadata immediately
+    return membersMetadata.get(localMember).block();
```
Comment (ronenhamias) [inline cluster/src/main/java/io/scalecube/cluster/metadata/MetadataStoreImpl.java]: using block is a good idea?

File: cluster/src/main/java/io/scalecube/cluster/metadata/MetadataStoreImpl.java [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Scheduler;
+
+public class MetadataStoreImpl implements MetadataStore {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(MetadataStoreImpl.class);
+
+  // Qualifiers
+
+  public static final String GET_METADATA_REQ = "sc/metadata/req";
```
[the remark this answers is not in the document]
Comment (artem-v) [reply cluster/src/main/java/io/scalecube/cluster/metadata/MetadataStoreImpl.java]: I know, but did it to be in sync with existing approach where such qualifier constants placed in implementations.

File: cluster/src/main/java/io/scalecube/cluster/metadata/MetadataStore.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+   */
+  void updateMetadata(Member member, Map<String, String> metadata);
+
+  /**
+   * Retrives cluster member metadata remotely.
+   *
+   * @param member cluster member
+   * @return mono result of getting remote member metadata
+   */
+  Mono<Map<String, String>> fetchMetadata(Member member);
```
[the remark this answers is not in the document]
Comment (artem-v) [reply cluster/src/main/java/io/scalecube/cluster/metadata/MetadataStore.java]: Updated javadoc for `metadata(Member)` and `member()`.  In short - `metadata(Member)` returns immediately what it has from _store_   while  the `fetch..`  is doing remote call for _dowloading_  cluster member metadata.

File: cluster/src/main/java/io/scalecube/cluster/metadata/MetadataStoreImpl.java [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+
+  @Override
+  public void updateMetadata(Member member, Map<String, String> metadata) {
+    Map<String, String> memberMetadata = new HashMap<>(metadata);
+    LOGGER.debug("Update member {} with new metadata: {}", member, memberMetadata);
+    membersMetadata.put(member, memberMetadata);
+  }
+
+  @Override
+  public void removeMetadata(Member member) {
```
[the remark this answers is not in the document]
Comment (artem-v) [reply cluster/src/main/java/io/scalecube/cluster/metadata/MetadataStoreImpl.java]: When we need to react on result of `remove..` then well, let's add some result, until that moment, let's keep it as `void`.

**Pull request merged**